web/cobrands/fixmystreet/fixmystreet.js

```diff
@@ -919,7 +930,13 @@ $.extend(fixmystreet.set_up, {
             // we want to record the navigation as a state, so the user
             // can return to it later using their Back button.
             if ('pushState' in history) {
-                history.pushState({ initial: true }, null, reportListUrl);
+                history.pushState({
+                    initial: true,
+                    mapState: around_map_state
+                }, null, reportListUrl);
+            }
+            if (around_map_state) {
+                fixmystreet.maps.set_map_state(around_map_state);
```

Comment:
Do we want to do this? This makes "Back to all reports" different to clicking Back (link resets the view, clicking Back does not) - and I wonder when you click "Back to all reports" do you mean "Back to the list of reports I was looking at" or "Back to the view of all reports". I think I lean towards the latter, so if you've scrolled the map when looking at individual reports, clicking "Back to all reports" would show me all the reports around where I now am on the map. Ooh and especially as "Problems nearby" does the same thing as "Back to all reports" and that definitely carries weight of it being map-view-currently-seen.

Reply:
Hmm, tricky. I can see advantages to both. I think I'd expect it to zoom back out and give an overview of all reports, based on the link text. Maybe it should do something different on `/reports` vs `/around`? On `/reports` it could restore the original map view which shows all reports, and `/around` would leave the current zoom level intact and show a list of nearby reports. (Let's ignore the third behaviour of "zoom in and show nearby reports" when clicked from a directly-loaded `/report` page!)

Reply:
I should note Google Maps invariably doesn't do what I want in this area, always seems to be messing up going back stuff :) I think I'd rather avoid different behaviour between pages, unless we're pretty sure that's better for some reason, I can see both being wanted on `/reports` too.

I think I'm happy for it to do what you suggest for `/reports` everywhere, but only if the Back button does the same thing (you are going Back to the previous view, as it were). (I assume your other PR on using replaceState might affect this too, will clicking Back go to the first/last replacedState thing...), so that would mean having the initial state save also save the map position so it can be used when going Back there. And if "problems nearby" then always did the `/around` behaviour you suggest, as that seems clear, and lets people (should they manage to work it out :) ) choose what they want.

Reply:
From our conversation on Slack:

> * Clicking "Back to all reports" or clicking Back (at the point that would be going back to all reports IYSWIM, not to another report): should zoom out and reset location to as it was. This means having the initial state save also save the map position so it can be used when clicking Back;
> * "Problems nearby" should NOT move map, but stay where you are, presumably also keeping the same zoom.

I've implemented these changes.